src/ioc_scan/stix_extract.py

```diff
@@ -0,0 +1,143 @@
+    urls = []
+
+    # Define hash type priority. Lower value means higher priority.
+    hash_priority = OrderedDict([("SHA256", 0), ("SHA1", 1), ("MD5", 2)])
+
+    # Iterate over each indicator in the STIX package
+    for indicator in stix_package.indicators:
+        for observable in indicator.observables:
+            object_type = observable.object_.properties._XSI_TYPE
+            if object_type == "AddressObjectType":
+                # Convert cybox.common.properties.String to str
+                ip_addresses.append(str(observable.object_.properties.address_value))
+            elif object_type == "FileObjectType":
+                hashes_dict = observable.object_.properties.hashes
+                if hashes_dict:
+                    best_hash = None
+                    best_priority = float("inf")
+                    for h in hashes_dict:
+                        if (
+                            h.type_.value in hash_priority
+                            and hash_priority[h.type_.value] < best_priority
+                        ):
+                            best_hash = str(h.simple_hash_value.value)  # Convert to str
+                            best_priority = hash_priority[h.type_.value]
+                    if best_hash is not None:
+                        hashes.append(best_hash)
+            elif object_type == "DomainNameObjectType":
+                # Convert cybox.common.properties.String to str
+                fqdns.append(str(observable.object_.properties.value.value))
+            elif object_type == "URIObjectType":
+                # Convert cybox.common.properties.String to str
+                urls.append(str(observable.object_.properties.value.value))
+
+    return ip_addresses, hashes, fqdns, urls
+
+
+def sort_ip_address(ip):
+    """
+    Take an IP address as input and return a tuple that can be used for sorting.
+
+    Args:
+        ip (str): an IP address
+
+    Returns:
+        tuple: a tuple containing two elements - the IP version (int) and the integer representation of the IP address (int).
+    """
+    ip_obj = ipaddress.ip_address(ip)
+    return (ip_obj.version, int(ip_obj))
+
+
+def main():
+    """Parse command line arguments and extract information from the STIX file."""
+    # Parse command line arguments
+    args = docopt(__doc__, version=__version__)
+    # Extract data from the STIX file or from stdin
+    stix_file = (
+        args["<file>"]
+        if args["<file>"]
+        else TextIOWrapper(sys.stdin.buffer, encoding="utf-8")
+    )
+    # Extract data from the STIX file
+    ip_addresses, hashes, fqdns, urls = extract_stix_info(stix_file)
+    # Sort IP addresses naturally (by their integer representation)
+    ip_addresses.sort(key=sort_ip_address)
+    hashes.sort()
+    fqdns.sort()
+    urls.sort()
+    # Print IPs, hashes, FQDNs, and URLs with separators and titles
+    print(f"\n{'#' * 20}\n# IP Addresses\n{'#' * 20}\n")
+    for ip in ip_addresses:
+        print(ip)
```

Comment:
If we're going to have four instances of the same logic I think we should DRY this out into a function. All four data sources are iterables and we're printing the headers with identical formatting.